### PR TITLE
AO3-4250 Made gift recipients update on blurbs on pseud/username changes

### PR DESCRIPTION
**Pull request by Bilka2**
## Issue

https://otwarchive.atlassian.net/browse/AO3-4250

## Purpose

Made it so that changing a pseud or username, deleting a pseud, or deleting a user account will expire the cache for all gift works (so that the byline will no longer display the old pseud or username.)

For deleting a pseud, this is done by expiring the cache on all gifts for the default pseud (since gifts are reassigned to the default pseud.) This will clear more caches than necessary - technically we only need to expire the cache for the gifts that were previously assigned to pseud being deleted.

## References

Adopts #5063
PR limit does not apply to volunteers with write access to the repository.

## Credit

Stephen Lewis, he/him (original code)
Brian Austin (original review)
Bilka (original review)
Bilka

### CODE REVIEW

File: features/users/user_edit.feature [hunk 1]
```diff
@@ -210,21 +210,29 @@ Feature:
       And I should see "after" within "#main"
       And I should not see "before" within "#main"
 
-  Scenario: Changing username updates series blurbs
-    Given I have no users
-      And I am logged in as "oldusername" with password "password"
+  Scenario: Changing only username updates series blurbs
```
Comment (sarken) [inline features/users/user_edit.feature]: *blinks* This is a fascinating edge case. After reading through this test a few times, I understand why it works like this rather than putting the username-matching pseud on the work, but still... wow.

Since it did take me a couple read throughs to figure out what was going on here, could we tweak the description a bit? Something like `Changing username to match an existing non-default pseud updates series blurbs`

**Review by sarken — APPROVED** [review]
One tiny thing, not really blocking! I'm excited to hopefully put this series of bugs behind us; thank you for working on this 🎉

File: features/users/user_edit.feature (hunk 1 above)
Comment (Bilka2) [reply features/users/user_edit.feature]: My test name change was rooted in the same confusion, but yours is even clearer, updated!

File: features/other_a/pseud_delete.feature [hunk 2]
```diff
@@ -207,3 +207,28 @@ Feature: Delete pseud.
...
+      And I am logged in as "gifter"
+      And I set up the draft "GiftStory1"
+      And I give the work to "me2 (giftee1)"
+      And I press "Post"
+      And I am logged in as "giftee1"
+    When I go to giftee1's gifts page
+    Then I should see "GiftStory1 by gifter for me2 (giftee1)"
+    When "giftee1" deletes the pseud "me2"
+      And I go to giftee1's gifts page
+    Then I should see "GiftStory1 by gifter for giftee1"
```
Comment (sarken) [inline features/other_a/pseud_delete.feature:236]: It looks like tests failed on this line a couple times since the last push. 😭 It's hopefully just flaky, but I'm going to switch this from Ready to Merge to Needs Coder Action

**Review by sarken — CHANGES_REQUESTED** [review]

File: features/other_a/pseud_delete.feature (hunk 2 above)
Comment (Bilka2) [reply features/other_a/pseud_delete.feature:236]: Definitely [flaky](https://github.com/Bilka2/otwarchive/actions/runs/33556604844), plus [some others](https://github.com/Bilka2/otwarchive/actions/runs/33557755661) as well. All fixed now: [Pseuds](https://github.com/Bilka2/otwarchive/actions/runs/33557244971), [Users](https://github.com/Bilka2/otwarchive/actions/runs/33558407752)

**Review by sarken — APPROVED** [review]
🎉
